utils.py

```diff
@@ -17,3 +19,51 @@ async def fetch_plugin_json(url):
         response = await client.get(url)
         return response.json()
 
+
+def check_version_zip(name_plugin: str, version: int) -> bool:
+    """
+    Check if the cached plugin is updated.
+    
+    Returns:
+        False if is not updated or not exist (will be updated) otherwise return True
+    """
+    # Define a cache directory
+    cache_json = "zip_cache/" + "versions.json"
+    if not os.path.exists(cache_json):
+        with open(cache_json, "w") as json_file:
```

Comment:
You are reading the file twice, I suggest:
- if the file does not exist, do not write an empty one, just initialize `versions = []`
- if it exists, open it and assign `versions = ...` as you do in line 40